base/src/main/scala/data/ProCompose.scala

```diff
@@ -0,0 +1,54 @@
+package scalaz
+package data 
+
+object ProCompose {
+  import typeclass.{Profunctor, Strong, Category, StrongClass, CategoryClass}
+  import typeclass.Strong._
+  import typeclass.Profunctor._
+
+  trait ProCompose[P[_, _], A, B] { } 
+
+  case class Pure[P[_, _], A, B](pab: A => B) extends ProCompose[P, A, B]
+
+  case class Composed[P[_, _], A, X, B](pab: P[A, X], more: ProCompose[P, X, B]) extends ProCompose[P, A, B]
```

Comment:
Not sure what exactly `Semiarrow` is (google doesn't help either), but if it gives you at least `arr` and `compose`, then you can.

> I'm curious why you think this is either balanced composition

I didn't say this _is_ balanced composition, but that this _supports_ balanced composition. A better word would have been _allows_ or _admits_.